flink-java/src/main/java/org/apache/flink/api/java/operators/PartitionOperator.java

```diff
@@ -84,8 +84,8 @@ public PartitionOperator(DataSet<T> input, Keys<T> pKeys, Partitioner<?> customP
 		Preconditions.checkArgument(distribution == null || pMethod == PartitionMethod.RANGE, "Customized data distribution is only neccessary for range partition.");
 		
 		if (distribution != null) {
-			Preconditions.checkArgument(distribution.getNumberOfFields() == pKeys.getNumberOfKeyFields(), "The number of key fields in the distribution and range partitioner should be the same.");
-			Preconditions.checkArgument(Arrays.equals(distribution.getKeyTypes(), pKeys.getKeyFieldTypes()), "The types of key from the distribution and range partitioner are not equal.");
+			Preconditions.checkArgument(pKeys.getNumberOfKeyFields() <= distribution.getNumberOfFields(), "The number of key fields in range partitioner should be less than the number in the distribution.");
```

Comment:
Update the message to "The distribution must provide at least as many fields as flat key fields are specified."